### PR TITLE
Fix APEX build

### PR DESCRIPTION
**Pull request by msimberg**
Adds APEX temporarily as a dependency to the `threadmanager` module.

### CODE REVIEW

File: libs/threadmanager/CMakeLists.txt [hunk 1]
```diff
@@ -45,3 +45,9 @@ add_hpx_module(threadmanager
     hpx_topology
   CMAKE_SUBDIRS examples tests
 )
+
+if (TARGET hpx::apex)
+  target_link_libraries(hpx_threadmanager PRIVATE hpx::apex)
+  get_target_property(_apex_inc_dir hpx::apex INTERFACE_INCLUDE_DIRECTORIES)
+  target_include_directories(hpx_compute INTERFACE "${_apex_inc_dir}")
```
Comment (hkaiser) [inline libs/threadmanager/CMakeLists.txt]: Why not just add `hpx::apex` as a dependency to `hpx_compute` as well?

Comment (msimberg) [reply libs/threadmanager/CMakeLists.txt]: `hpx_compute` already depends on APEX. I'll add it to `hpx_compute_cuda` though.

File: libs/compute_cuda/CMakeLists.txt [hunk 2]
```diff
@@ -62,3 +62,9 @@ add_hpx_module(compute_cuda
     hpx_serialization
   CMAKE_SUBDIRS examples tests
 )
+
+if (TARGET hpx::apex)
+  target_link_libraries(hpx_threadmanager PRIVATE hpx::apex)
```
Comment (hkaiser) [inline libs/compute_cuda/CMakeLists.txt]: Is this what you intended? Shouldn't this be `hpx_compute_cuda` instead?

Comment (msimberg) [reply libs/compute_cuda/CMakeLists.txt]: Uh, no... Thanks!

**Comment by msimberg** [comment]
@hkaiser I removed the include directories. Things seem to be happy without them. They were probably a leftover from somewhere else.

**Comment by msimberg** [comment]
Fixed in #4214.

**Pull request closed without merging**